cli/flox/doc/flox-config.md

```diff
@@ -7,69 +7,111 @@ header: "flox User Manuals"
+```
+flox [ <general-options> ] config
+     [-l |
+      -r |
+      --set <key> <string> |
+      --set-number <key> <number> |
+      --set-bool <key> <bool> |
+      --delete=<key>]
+```
 
 # DESCRIPTION
 
-Configure and/or display user-specific parameters.
+Without any flags or when `-l` is passed, `flox config` shows all options with
+their computed value.
+
+Config values are read from the following sources in order of descending priority:
+
+1. Environment variables.
+   All config options may be set by prefixing with `FLOX_` and using
+   SCREAMING_SNAKE_CASE.
+   For example, `disable_metrics` may be set with `FLOX_DISABLE_METRICS=true`.
+1. User customizations from `$FLOX_CONFIG_HOME/flox.toml` if set or else
+   `$XDG_CONFIG_HOME/flox/flox.toml`.
+1. User customizations from `flox/flox.toml` in any of `$XDG_CONFIG_DIRS`.
+1. System settings from `/etc/flox.toml`.
+1. `flox` provided defaults.
 
-Without any flags or `-l` or `--list` shows all configurable options
-with their computed value.
+`flox config` commands that mutate configuration always write to
+`${FLOX_CONFIG_HOME:-$XDG_CONFIG_HOME}/flox/flox.toml`.
 
 ## Key Format
 
-`<key>` supports dot-separated queries for nested vaules, e.g.
+`<key>` supports dot-separated queries for nested values, for example:
 
 ```
-flox config --set 'nix.access_tokens."github.com"' "ghp_xxx"`
+flox config --set 'trusted_environments."owner/name"' trust
 ```
 
-## Config Format
+# OPTIONS
 
-All config keys can be listed with
+## Config Options
 
-```
-flox config
-```
+`-l`, `--list`
+:   List the current values of all options.
 
-The dispalyed values are resolved by reading:
+`-r`, `--reset`
+:   Reset all options to their default values without confirmation.
 
-1. package defaults from `$PREFIX/etc/flox.toml`
-2. installation defaults from `/etc/flox.toml`
-3. user customizations from `$HOME/.config/flox/flox.toml`
-4. environment variables
+`--set <key> <string>`
+:  Set `<key> = <string>` for string values
 
-where the last occurence is used as the final value.
+`--set-number <key> <number>`
+:  Set `<key> = <number>` for number values
 
-`flox config` commands that mutate configurations always write to `$HOME/.config/flox/flox.toml`.
+`--set-bool <key> <bool>`
+:  Set `<key> = <bool>` for boolean values
 
-# OPTIONS
+`--delete <key>`
+:   Delete config key
 
 ```{.include}
 ./include/general-options.md
-./include/development-options.md
 ```
 
-## Config Options
+# SUPPORTED CONFIGURATION OPTIONS
+
+`config_dir`
+:   Directory where flox should load its configuration file (default:
+    `$XDG_CONFIG_HOME/flox`).
+    This option will only take effect if set with `$FLOX_CONFIG_HOME`.
+    `$FLOX_CONFIG_DIR` and `config_dir` are ignored.
+
+`cache_dir`
+:   Directory where flox should store ephemeral data (default:
+    `$XDG_CACHE_HOME/flox`).
+
+`data_dir`
+:   Directory where flox should store persistent data (default:
+    `$XDG_DATA_HOME/flox`).
```

Comment:
suggestion(nb): looks more ordered in the source view, not that it matters once rendered

```suggestion
:   Directory where flox should store ephemeral data 
    (default: `$XDG_CACHE_HOME/flox`).

`data_dir`
:   Directory where flox should store persistent data
    (default: `$XDG_DATA_HOME/flox`).
```